File: services/inventory-service/rabbitmq/rabbit.go
package rabbitmq

import (
	"fmt"
	"log"

	"github.com/rabbitmq/amqp091-go"
	amqp "github.com/rabbitmq/amqp091-go"
)

type rabbit struct {
	User     string
	Password string
	Host     string
	Port     int
	Conn     *amqp091.Connection
}

func NewRabbit(user string, port int, pass string, host string) *rabbit {

	return &rabbit{
		User:     user,
		Password: pass,
		Host:     host,
		Port:     port,
		Conn:     nil,
	}
}

func (r *rabbit) Conection() error {
	url := fmt.Sprintf("amqp://%s:%s@%s:%d/",
		r.User,
		r.Password,
		r.Host,
		r.Port,
	)

	conn, err := amqp.Dial(url)
	if err != nil {
		return fmt.Errorf("falha ao conectar no RabbitMQ em %s:%d: %w", r.Host, r.Port, err)
	}

	r.Conn = conn

	log.Printf("Conectado ao RabbitMQ em %s:%d", r.Host, r.Port)

	return nil
}

File: services/order-service/rabbitmq/rabbit.py
from pydantic import BaseModel
import pika
class Rabbit(BaseModel):
    host:str
    user:str
    password:str
    port: int

    # def __init__(self, host, user, password):
    #     self.host=host
    #     self.password=password
    #     self.user=user
    def conection(self):
        connection = pika.BlockingConnection(pika.URLParameters(f'amqp://{self.user}:{self.password}@{self.host}:{self.port}/'))
        
        return connection